### ISSUE TITLE
[MSB4018] The "ResolvePackageDependencies" task failed unexpectedly

### ISSUE THREAD

**Issue by ValentijnMakkenze** [firegiant/HeatWaveSupport#10]
### HeatWave Version

0.9.0.5

### Visual Studio Version

17.4.2

### Repro Steps

1. I converted two setup projects in my solution
2. Both give an MSB4018 when the solution is loaded

### Actual Result

5-12-2022 08:39:06	Critical	Project 'Neuron.Brk.Monitor.InstanceManager.Setup' load failed|   [MSB4018] The "ResolvePackageDependencies" task failed unexpectedly.
System.ArgumentException: Invalid framework version '0.0/win'.
   at NuGet.Frameworks.NuGetFramework.ParseFrameworkNameParts(IFrameworkNameProvider mappings, String[] parts, String& framework, Version& version, String& profile)
   at NuGet.Frameworks.NuGetFramework.ParseFrameworkName(String frameworkName, IFrameworkNameProvider mappings)
   at NuGet.Frameworks.NuGetFramework.Parse(String folderName, IFrameworkNameProvider mappings)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.<GetPackageAndFileDefinitions>b__48_5(String tg)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.Contains[TSource](IEnumerable`1 source, TSource value, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.Contains[TSource](IEnumerable`1 source, TSource value)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.<>c__DisplayClass48_1.<GetPackageAndFileDefinitions>b__3(IAssetsLogMessage log)
   at System.Linq.Enumerable.WhereListIterator`1.MoveNext()
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.<GetPackageAndFileDefinitions>g__GetPackageDiagnosticLevel|48_0(LockFileLibrary package)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.GetPackageAndFileDefinitions()
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.ExecuteCore()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() at (185:5)

### Expected Result

No errrors

**Comment by FireGiantHelp** [comment]
We have not been able to reproduce this issue. Would it be possible to share your .wixproj and the projects it depends on? If it is not possible to share the projects, could you share exactly what type of projects are being used so we can try to reproduce the problem?

For example: 
* `Neuron.Brk.Monitor.InstanceManager.Setup.wixproj` - MSI WiX v4 project
* `Neuron.ExampleName.csproj` - .NET 6 WPF Application .csproj
* `Neuron.Helper.csproj` - .NET Standard 2.0 .csproj
* etc

**Comment by austiv32** [comment]
I believe this is related to specifically targeting windows. Here's a minimal reproduction:

**Wix4Test.Console.Setup.wixproj**
```
<Project Sdk="WixToolset.Sdk/4.0.0-preview.1">
  <ItemGroup>
    <ProjectReference Include="..\Wix4Test.Console\Wix4Test.Console.csproj" />
  </ItemGroup>
</Project>
```

**Wix4Test.Console.csproj** (Visual Studio's default console application template is sufficient)
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0-windows</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>
</Project>
```

**Error List in Visual Studio:**
```
Error	MSB4018	The "ResolvePackageDependencies" task failed unexpectedly.
System.ArgumentException: Invalid framework version '0.0/win'.
   at NuGet.Frameworks.NuGetFramework.ParseFrameworkNameParts(IFrameworkNameProvider mappings, String[] parts, String& framework, Version& version, String& profile)
   at NuGet.Frameworks.NuGetFramework.ParseFrameworkName(String frameworkName, IFrameworkNameProvider mappings)
   at NuGet.Frameworks.NuGetFramework.Parse(String folderName, IFrameworkNameProvider mappings)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.<GetPackageAndFileDefinitions>b__48_5(String tg)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.Contains[TSource](IEnumerable`1 source, TSource value, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.Max[TSource](IEnumerable`1 source)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.<GetPackageAndFileDefinitions>g__GetPackageDiagnosticLevel|48_0(LockFileLibrary package)
   at FireGiant.HeatWave.Build.Tasks.ResolvePackageDependencies.GetPackageAndFileDefinitions()
   at Microsoft.NET.Build.Tasks.TaskBase.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()	Wix4Test.Console.Setup	C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\FireGiant\HeatWave\FireGiant.HeatWave.PackageDependencyResolution.targets	185
```

The build does complete successfully in Visual Studio, but the implicit restore fails. Build output:
```
Build started...
1>------ Build started: Project: Wix4Test.Console, Configuration: Debug Any CPU ------
Restored C:\<path-to>\Wix4Test\Wix4Test.Console\Wix4Test.Console.csproj (in 1 ms).
Failed to restore C:\<path-to>\Wix4Test\Wix4Test.Console.Setup\Wix4Test.Console.Setup.wixproj (in 2 ms).
NuGet package restore failed. Please see Error List window for detailed warnings and errors.
1>Wix4Test.Console -> C:\<path-to>\Wix4Test\artifacts\Wix4Test.Console.dll
2>------ Build started: Project: Wix4Test.Console.Setup, Configuration: Debug x64 ------
2>Wix4Test.Console.Setup -> C:\<path-to>\Wix4Test\artifacts\Wix4Test.Console.Setup.msi
========== Build: 2 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
========== Elapsed 00:02.026 ==========
```

Building from command-line with dotnet fails with the following output:
```
$ dotnet.exe build Wix4Test.sln
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
C:\<path-to>\Wix4Test\Wix4Test.Console.Setup\Wix4Test.Console.Setup.wixproj : error NU1201: Project Wix4Test.Console is not compatible with native (native,Version=v0.0). Project Wix4Test.Console supports: net6.0-windows7.0 (.NETCoreApp,Version=v6.0) [C:\<path-to>\Wix4Test\Wix4Test.sln]
  Failed to restore C:\<path-to>\Wix4Test\Wix4Test.Console.Setup\Wix4Test.Console.Setup.wixproj (in 77 ms).
  1 of 2 projects are up-to-date for restore.

Build FAILED.

C:\<path-to>\Wix4Test\Wix4Test.Console.Setup\Wix4Test.Console.Setup.wixproj : error NU1201: Project Wix4Test.Console is not compatible with native (native,Version=v0.0). Project Wix4Test.Console supports: net6.0-windows7.0 (.NETCoreApp,Version=v6.0) [C:\<path-to>\Wix4Test\Wix4Test.sln]
    0 Warning(s)
    1 Error(s)
```
Subsequent dotnet builds with the --no-restore flag succeed.

When changing TargetFramework in Wix4Test.Console.csproj to net6.0 (instead of net6.0-windows), everything builds fine, both in Visual Studio and on command-line with dotnet.

**Comment by FireGiantHelp** [comment]
@austiv32 Thank you for the additional detail. Your issue sounds like wixtoolset/issues#7034 which should be resolved in wix4-rc.1 coming soon.

@ValentijnMakkenze We tested with .NET Framework v4.8 projects and were not able to reproduce the issue. Could you attach the `[YourWixprojFolder]\obj\project.assets.json` file? The dev team thinks that file should help them figure out how to reproduce the issue. If you do not feel comfortable attaching the file here, could you email it to us privately at `support at firegiant.com`?

**Comment by ValentijnMakkenze** [comment]
[project.assets.json.txt](https://github.com/firegiant/HeatWaveSupport/files/10231161/project.assets.json.txt)
Had to rename it to .txt as .json isn't supported.

**Comment by austiv32** [comment]
@FireGiantHelp Thanks, the workaround on the WiX issue cleared up the MSB4018 in addition to NU1201 dotnet build issues

**Comment by FireGiantHelp** [comment]
We still have not been able to reproduce the issue described by @ValentijnMakkenze but believe this may be a symptom of a more complex issue with project references that we are investigating. We'll keep this issue open as the investigation continues.

**Comment by FireGiantHelp** [comment]
We believe we have a fix for this. It is dependent on changes coming in WiX v4.0.0-rc.2 so look for a HeatWave update after WiX v4-rc.2 is available.

**Comment by FireGiantHelp** [comment]
We expect this issue is fixed in HeatWave v0.9.4 now available in Visual Studio Marketplace so Visual Studio should provide notification of the update.

Note: HeatWave v0.9.4 requires WiX v4-rc.2 so be sure to update any existing WiX v4 projects to -rc.2 when upgrading to this latest HeatWave.
